src/nupic/algorithms/Connections.hpp

```diff
@@ -41,11 +41,11 @@ namespace nupic {
 namespace algorithms {
 namespace connections {
 
-typedef UInt32 CellIdx;
-typedef UInt16 SegmentIdx; /** Index of segment in cell. */
-typedef UInt16 SynapseIdx; /** Index of synapse in segment. */
-typedef UInt32 Segment;    /** Index of segment's data. */
-typedef UInt32 Synapse;    /** Index of synapse's data. */
+typedef UInt32 CellIdx;  //TODO instead of typedefs, use templates for proper type-checking
+typedef unsigned char SegmentIdx; /** Index of segment in cell. */
+typedef unsigned char SynapseIdx; /** Index of synapse in segment. */
```

Comment:
@ctrl-z-9000-times I replaced lot of plain UInts with these typedefs, can you please double-check I use them as intended? (ei SynapseIdx and not Synapse, etc..)

Reply:
I read through this and I think you've got all of the types right.

You need to add checks to the methods `Connections::addSynapse` & `Connections::addSegment` to check for too many synapses/segments.  Otherwise it will fail silently.

Reply:
> You need to add checks to the methods Connections::addSynapse & Connections::addSegment to check for too many synapses/segments. Otherwise it will fail silently. 

Great point! I had only the option to detect overflow in Debug